server/service/src/processors/mod.rs

```diff
@@ -0,0 +1,110 @@
+};
+
+#[cfg(test)]
+mod test_helpers;
+pub(crate) mod transfer;
+
+const NUMBER_OF_CHANNELS: usize = 30;
+
+#[derive(Clone)]
+pub struct ProcessorsTrigger {
+    requisition_transfer: Sender<()>,
+    shipment_transfer: Sender<()>,
+}
+
+pub struct Processors {
+    requisition_transfer: Receiver<()>,
+    shipment_transfer: Receiver<()>,
+}
+
+#[derive(Debug, Error)]
+enum ProcessorsError {
+    #[error("Error in shipment transfer processor ({0})")]
+    ShipmentTransfer(ProcessShipmentTransfersError),
+    #[error("Error in requisition transfer processor ({0})")]
+    RequisitionTransfer(ProcessRequisitionTransfersError),
+}
+
+impl Processors {
+    pub fn init() -> (ProcessorsTrigger, Processors) {
+        let (requisition_transfer_sender, requisition_transfer_receiver) =
+            mpsc::channel(NUMBER_OF_CHANNELS);
+
+        let (shipment_transfer_sender, shipment_transfer_receiver) =
+            mpsc::channel(NUMBER_OF_CHANNELS);
+
+        (
+            ProcessorsTrigger {
+                requisition_transfer: requisition_transfer_sender,
+                shipment_transfer: shipment_transfer_sender,
+            },
+            Processors {
+                requisition_transfer: requisition_transfer_receiver,
+                shipment_transfer: shipment_transfer_receiver,
+            },
+        )
+    }
+
+    pub fn spawn(self, service_provider: Arc<ServiceProvider>) -> JoinHandle<()> {
+        let Processors {
+            mut requisition_transfer,
+            mut shipment_transfer,
+        } = self;
+
+        tokio::spawn(async move {
+            loop {
+                let result = tokio::select! {
+                    Some(_) = requisition_transfer.recv() => {
+                        process_requisition_transfers(&service_provider).map_err(ProcessorsError::RequisitionTransfer)
+                    },
+                    Some(_) = shipment_transfer.recv() => {
+                        process_shipment_transfers(&service_provider).map_err(ProcessorsError::ShipmentTransfer)
+                    },
+                    else => break,
+                };
+
+                if let Err(error) = result {
+                    log::error!("{}", error);
+                }
+            }
+        })
+    }
+}
+
+impl ProcessorsTrigger {
+    pub(crate) fn trigger_requisition_transfers(&self) {
+        if let Err(error) = self.requisition_transfer.try_send(()) {
+            log::error!(
+                "Problem triggering requisition transfer processor {:#?}",
+                error
+            )
+        }
+    }
+
+    pub(crate) fn trigger_shipment_transfers(&self) {
+        if let Err(error) = self.shipment_transfer.try_send(()) {
+            log::error!(
+                "Problem triggering shipment transfer processor {:#?}",
+                error
+            )
+        }
+    }
+
+    pub(crate) fn new_test() -> ProcessorsTrigger {
+        ProcessorsTrigger {
+            requisition_transfer: mpsc::channel(1).0,
```

Comment:
What I was trying to do with this:
Ability to create ServiceProvider for test that don't need to worry about processors, while having non optional `processors_trigger` in ServiceContext and ServiceProvider.

> Also, Is there a reason for doing a buffer length 1 here rather than the NUMBER_OF_CHANNELS?

I thought since it's going to be discarded anyway, maybe use minimum number for buffer.

> With a different number of buffered items here there's potentially different behaviour in testing vs production code, would rather not have that inconsistency...

> With a buffer length of 1 if more than one trigger is created by a test, would that mean it would just block as there's no receiver?

Basically with `ProcessorsTrigger::new_test()`, the sender is discarded, so the channel trigger is unusable (it will log an error when triggered). 

I can't think of a better name `mock` doesn't quite work as it's not meant to be usable, would a comment be sufficient ?

Reply:
Yeah a comment is totally fine. I think the main point is that it needs a comment as it's not clear :)

Haven't thought of a brilliant name but maybe `new_with_no_triggers` or `new_with_no_processors` would be more descriptive?

Reply:
`new_pointless` ? hehe.

I've added a change straight to head branch: 5a8539fba7e5f6bb7a27702aa5771b1b207ae23e
* ProcessorTriggers::new_void() // with comment
* And other instantiations that use this method are `_with_no_processors`

Reply:
Excellent. Much clearer!